test/sr_key_set_test.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2024 Codership Oy <info@codership.com>
+ *
+ * This file is part of wsrep-lib.
+ *
+ * Wsrep-lib is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Wsrep-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with wsrep-lib.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "wsrep/key.hpp"
+#include "wsrep/sr_key_set.hpp"
+#include <boost/test/unit_test.hpp>
+
+static void append_key_parts(wsrep::key& key,
+                             const std::vector<std::string>& parts)
+{
+    for (const std::string& part : parts)
+    {
+        key.append_key_part(part.c_str(), part.length());
+    }
+}
+
+BOOST_AUTO_TEST_CASE(sr_key_set_test_contains)
+{
+    wsrep::sr_key_set key_set;
+
+    { // contains same key
```

Comment:
It would be better to have separate test cases named according to what they test than having multiple blocks inside single test case. If the test fails, it it much easier to see immediately what was the testcase which failed.

If there is some common setup step for all the tests, it can be put into test fixture.